locations/spiders/mcdonalds_mt.py

```diff
@@ -4,127 +4,37 @@
-        else:
-            for day_group in day_groups:
-                if day_group["from_day"] == day_group["to_day"]:
-                    opening_hours += "{from_day} {hours}; ".format(**day_group)
-                else:
-                    opening_hours += "{from_day}-{to_day} {hours}; ".format(**day_group)
-            opening_hours = opening_hours[:-2]
-
-        return opening_hours
-
-    def parse_address(self, response):
-        match = re.search(r"<h3>([\s|\S]{1,})</h3>", response)
-        if not match:
-            return ""
-        data = match.groups()[0]
-        ref, address = data.split("-")
-        return ref.strip(), address.strip()
-
-    def parse_latlon(self, response):
-        match = re.search(r"',(.*),(.*),map", response)
-        lat, lon = match.groups()
-        return lat, lon
-
-    def parse_phone(self, response):
-        match = re.search(r"</b>([\s|\S]{1,})</h4>", response)
-        if not match:
-            return ""
-        phone = match.groups()[0]
-        return phone.strip()
+    sitemap_urls = ["https://mcdonalds.com.mt/location-sitemap.xml"]
+    sitemap_rules = [(r"\/location\/([\w\-]+)\/", "parse")]
 
+    
     def parse(self, response):
-        match = re.search(r"placeMarkers\(\) {([\s|\S]{0,})}", response.text)
-        if not match:
-            return
-        stores = match.groups()[0]
-        stores = stores.split(";")
-        for store in stores:
-            store = store.strip()
-            if not store:
-                continue
-            ref, address = self.parse_address(store)
-            phone = self.parse_phone(store)
-            lat, lon = self.parse_latlon(store)
-            properties = {
-                "ref": ref,
-                "addr_full": address,
-                "phone": phone,
-                "name": "McDonalds",
-                "lat": lat,
-                "lon": lon,
-            }
-
-            yield Feature(**properties)
+        properties = {
+            "ref": re.search(self.sitemap_rules[0][0], response.url).group(1),
+            "name": response.xpath('//h1/text()').get().strip(),
+            "addr_full": response.xpath('//div/p/text()').get().strip(),
+            "lat": response.xpath('//div[@class="location-map"]/div[@id="map"]/@data-lat').get(),
+            "lon": response.xpath('//div[@class="location-map"]/div[@id="map"]/@data-lng').get(),
+            "website": response.url,
+        }
+
+        oh = OpeningHours()
+        for day in response.xpath('//div[@class="elementor-tab-content elementor-clearfix"]/table[@class="location-hours-table"]//tr')[0:6]:
+            oh.add_range(
+                day=day.xpath("./td[1]/text()").get(),
+                open_time=day.xpath("./td[2]/text()").get().split(" - ")[0],
+                close_time=day.xpath("./td[2]/text()").get().split(" - ")[1],
+                time_format="%H:%M",
+            )
+
+        properties["opening_hours"] = OpeningHours()
```

Comment:
There's no need to set this to a blank opening hours object if you're going to set it to something else on the next line.

Reply:
```suggestion
```